pkg/rbd/controllerserver.go

```diff
@@ -391,18 +391,24 @@ func (cs *ControllerServer) DeleteVolume(ctx context.Context, req *csi.DeleteVol
 
 	// Deleting rbd image
 	klog.V(4).Infof(util.Log(ctx, "deleting image %s"), rbdVol.RbdImageName)
-	if err := deleteImage(ctx, rbdVol, cr); err != nil {
+	if err = deleteImage(ctx, rbdVol, cr); err != nil {
 		klog.Errorf(util.Log(ctx, "failed to delete rbd image: %s/%s with error: %v"),
 			rbdVol.Pool, rbdVol.RbdImageName, err)
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if err := undoVolReservation(ctx, rbdVol, cr); err != nil {
+	if err = undoVolReservation(ctx, rbdVol, cr); err != nil {
 		klog.Errorf(util.Log(ctx, "failed to remove reservation for volume (%s) with backing image (%s) (%s)"),
 			rbdVol.RequestName, rbdVol.RbdImageName, err)
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	if rbdVol.Encrypted {
```

Comment:
just a re-cap of offline discussion:
yes, delete passphrase does not work, so we need to figure out where to put data about what kms was used and if image was encrypted

* option one is to use extra meta-data field for kms type - it will allow to keep track of where passphrase is with the image itself, the downside is it will require extra request to ceph to get each meta-data field (so in proposed scenario - 1st call to check if encrypted + 2nd to find kms type)
* option two - extend omapval object with new fields - this will require a way to handle old version of the object and new... I have not yet investigated what this will involve

Reply:
@vasyl-purchel I suggest we go ahead with option two. (like, the image name extension in the PR https://github.com/ceph/ceph-csi/pull/758 ).
- One consideration for storing extra attributes in the image itself, was it extends well to mirrored rbd images, but for such cases we need to anyway extend mirroring the vault keys as well. So this does not seem to yield a large benefit yet.
- The OMap for the image has only 3 keys (if we include the image naming PR and snapshot source), so we are not yet stressing the OMap key limits IMO and can add some more (for example I am thinking of adding the pool [parameter](https://github.com/ceph/ceph-csi/pull/760#discussion_r368085409) for topology support as well).

So keeping various moving parts with the same design idea, seems a better option at present.

Reply:
Ack, will update review once it's implemented

Reply:
this is implemented - new key in rados object `csi.volume.encryptKMS` with `<kms-type>|<kms-id>`. If not found or empty - treated as not encrypted with KMS hence no need for passphrase clean up